lib/Models/GlobeOrMap.js

```diff
@@ -246,12 +246,14 @@ GlobeOrMap.prototype._highlightFeature = function(feature) {
                   'marker-color': this.terria.baseMapContrastColor
                 };
 
+                var that = this;
                 var removeCallback = this._removeHighlightCallback = function() {
-                    catalogItem._hide();
-                    catalogItem._disable();
+                    that._highlightPromise.then(function() {
+                        catalogItem._hide();
+                        catalogItem._disable();
```

Comment:
I guess another option is to just catch the exception. That's not too much different from the current situation, but it does at least make it clear that the situation is expected and doesn't break anything. :)